src/Libraries/Microsoft.Extensions.Telemetry/Buffering/ILoggingBufferProvider.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if NET9_0_OR_GREATER
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Shared.DiagnosticIds;
+
+namespace Microsoft.Extensions.Logging;
+
+/// <summary>
+/// Interface providing access to the current logging buffer.
+/// </summary>
+[Experimental(diagnosticId: DiagnosticIds.Experiments.Telemetry, UrlFormat = DiagnosticIds.UrlFormat)]
+public interface ILoggingBufferProvider
```

Comment:
Thanks for such a nice comment!

I tried to implement this, but there was a small issue (and hopefully I got your design right) - `ExtendedLogger` would hold references to buffers which existed when every instance of `ExtendedLogger` was being created, which means it can't hold references to HttpRequest buffers because they have`Scoped` lifetime, while `ExtendedLogger` is of `Singleton` lifetime.

So I changed it that buffers are created on demand in the `GlobalBufferManager`/`HttpRequestBufferManager`. In that case, we don't need to expose `CreateBuffer()` methods.

The overall design looks like that now (simplified):
```
interface ILoggingBuffer
{
    TryEnqueue(event data)
    Flush()
    TruncateOverlimit()
}

interface IBufferSink
{
    LogRecords<T>(IEnumerable<T> serializedRecords)
}
interface IBufferManager
{
    TryEnqueue(IBufferSink sink)
    Flush()
}
interface IHttpRequestBufferManager : IBufferManager
{
    FlushCurrentRequestLogs()
}
```

`ExtendedLogger` now works with BufferManagers and also responsible for creating `IBufferSink`s. The canonical use case is as follows: inject either `IBufferManager` in any class or `IHttpRequestBufferManager` in a `Middleware`, and call respective` Flush*() `methods when required.